modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/util/NumericDDMFormFieldTypeUtil.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.form.field.type.internal.util;
+
+import com.liferay.dynamic.data.mapping.model.DDMFormField;
+import com.liferay.dynamic.data.mapping.render.DDMFormFieldRenderingContext;
+import com.liferay.dynamic.data.mapping.util.NumericDDMFormFieldUtil;
+import com.liferay.portal.kernel.json.JSONException;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * @author Carolina Barbosa
+ */
+public class NumericDDMFormFieldTypeUtil {
+
+	public static Map<String, Object> getParameters(
+		String dataType, DDMFormField ddmFormField,
+		DDMFormFieldRenderingContext ddmFormFieldRenderingContext) {
+
+		boolean inputMask = GetterUtil.getBoolean(
+			DDMFormFieldTypeUtil.getPropertyValue(
+				ddmFormField, ddmFormFieldRenderingContext, "inputMask"));
+		Locale locale = ddmFormFieldRenderingContext.getLocale();
+
+		if (!inputMask) {
+			if (!StringUtil.equals(dataType, "double")) {
+				return new HashMap<>();
+			}
+
+			return HashMapBuilder.<String, Object>put(
+				"symbols", _getSymbols(locale)
+			).build();
+		}
+
+		if (!StringUtil.equals(dataType, "double")) {
+			return HashMapBuilder.<String, Object>put(
+				"inputMask", inputMask
+			).put(
+				"inputMaskFormat",
+				DDMFormFieldTypeUtil.getPropertyValue(
+					ddmFormField, ddmFormFieldRenderingContext, locale,
+					"inputMaskFormat")
+			).build();
+		}
+
+		String numericInputMask = DDMFormFieldTypeUtil.getPropertyValue(
+			ddmFormField, ddmFormFieldRenderingContext, locale,
+			"numericInputMask");
+
+		return HashMapBuilder.<String, Object>put(
+			"inputMask", inputMask
+		).put(
+			"numericInputMask", numericInputMask
+		).putAll(
+			_getNumericInputMaskParameters(numericInputMask)
+		).build();
+	}
+
+	private static Map<String, Object> _getNumericInputMaskParameters(
+		String numericInputMask) {
+
+		try {
+			JSONObject numericInputMaskJSONObject =
+				JSONFactoryUtil.createJSONObject(numericInputMask);
+
+			return HashMapBuilder.<String, Object>put(
```

Comment:
I think it's possible to just deserialize the string using `JSONFactoryUtil.deserialize(String json)`. Could you please check that?

Reply:
It works 🎉

Reply:
![power](https://user-images.githubusercontent.com/11450348/147287098-27148d06-0162-41e3-b319-4eb85994a9ba.gif)